test/models/user_test.rb

```diff
@@ -228,4 +240,40 @@
       locks.each { |lock| lock.reload.deleted_at.wont_be_nil }
     end
   end
+
+  describe "#admin_for_project?" do
+    it "is true for a user that has been granted the role of project admin" do
+      users(:deployer_project_admin).is_admin_for?(projects(:test)).must_equal(true)
+    end
+
+    it "is false for users that have not been granted the role of project admin" do
+      users(:viewer).is_admin_for?(projects(:test)).wont_equal(true)
+      users(:deployer).is_admin_for?(projects(:test)).wont_equal(true)
+      users(:admin).is_admin_for?(projects(:test)).wont_equal(true)
+      users(:super_admin).is_admin_for?(projects(:test)).wont_equal(true)
+    end
+  end
+
+  describe "#deployer_for_project?" do
+    it "is true for a user that has been granted the role of project deployer" do
+      users(:viewer_project_deployer).is_deployer_for?(projects(:test)).must_equal(true)
+    end
+
+    it "is true for a user that has been granted the role of project admin" do
+      users(:deployer_project_admin).is_deployer_for?(projects(:test)).must_equal(true)
+    end
+
+    it "is false for users that have not been granted the roles of project deployer or project admin" do
+      users(:viewer).is_deployer_for?(projects(:test)).wont_equal(true)
+      users(:deployer).is_deployer_for?(projects(:test)).wont_equal(true)
+      users(:admin).is_deployer_for?(projects(:test)).wont_equal(true)
+      users(:super_admin).is_deployer_for?(projects(:test)).wont_equal(true)
```

Comment:
Hum, this test is a little confusing in that matter, now that I think about it.

System-level roles and Project-specific roles have been completely decoupled. The rights for deploying a project are checked in the ProjectLevelAuthorization module:
```
def authorize_project_deployer!
    unauthorized! unless current_user.is_deployer? || current_user.is_deployer_for?(current_project)
  end
```
Both system-level admins and super-admins will also be system-level deployers, which means, they will have deployer access to a project.

This being said, all I'm asserting in this test is that, regardless of the system-level role the user has, the method `User#is_deployer_for?(project)` will only return true if the User has a project-specific 'Deployer' role for that project, in the table `UserProjectRole`.